src/lib.rs

```diff
@@ -78,3 +84,62 @@ pub fn from_value<T: serde::de::DeserializeOwned>(value: JsValue) -> Result<T> {
 pub fn to_value<T: serde::ser::Serialize + ?Sized>(value: &T) -> Result<JsValue> {
     value.serialize(&Serializer::new())
 }
+
+/// A wrapper around a [`JsValue`] (or anything that can be cast into one)
+/// that makes it pass through serialization and deserialization unchanged.
+///
+/// # Example
+/// ```rust
+/// use serde_wasm_bindgen::{PreservedValue, to_value};
+/// #[derive(serde::Serialize)]
+/// struct MyStruct {
+///     int_field: i32,
+///     js_field: PreservedValue<JsValue>,
+/// }
+/// let big_array = js_sys::Int8Array::new_with_length(1000);
+/// let s = MyStruct {
+///     int_field: 5,
+///     js_field: PreservedValue(big_array.into()),
+/// };
+///
+/// // Will return a JsValue representing an object with two fields (`int_field` and `js_field`).
+/// // `js_field` will be an `Int8Array` pointing to the same underlying JavaScript object as `big_array`.
+/// to_value(&s);
+/// ```
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+#[serde(
+    into = "PreservedValueWrapper",
+    try_from = "PreservedValueWrapper",
+    bound = "T : JsCast + Clone"
+)]
+pub struct PreservedValue<T: JsCast + Clone>(pub T);
+
+// Some arbitrary string that no one will collide with unless they try.
+pub(crate) const PRESERVED_VALUE_MAGIC: &str = "1fc430ca-5b7f-4295-92de-33cf2b145d38";
+
+#[derive(serde::Serialize, serde::Deserialize)]
```

Comment:
Right now we only compile with Serde's derive feature in development mode. That's why tests pass, but using serde-wasm-bindgen after this PR in external project probably would fail with a compilation error.

The dependency with the derive feature needs to be moved in Cargo.toml to regular deps.

Reply:
Oh, good catch -- I've updated this. Do you think it's worth putting it behind a feature to reduce dependencies?

Reply:
Nah, I very much doubt someone uses serde-wasm-bindgen without derive - at that point might as well use `JsValue` directly.